Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -374,18 +410,65 @@ open class NavigationMapView: UIView {
         
         mostRecentUserCourseViewLocation = location
         
-        // While animating to overview mode, don't animate the puck.
-        let duration: TimeInterval = animated && navigationCamera.state != .transitionToOverview ? 1 : 0
-        UIView.animate(withDuration: duration, delay: 0, options: [.curveLinear]) { [weak self] in
-            guard let screenCoordinate = self?.mapView.screenCoordinate(for: location.coordinate) else { return }
-            self?.userCourseView.center = CGPoint(x: screenCoordinate.x, y: screenCoordinate.y)
+        switch puckType {
+        case .courseView(configuration: _):
+            // While animating to overview mode, don't animate the puck.
+            let duration: TimeInterval = animated && navigationCamera.state != .transitionToOverview ? 1 : 0
+            UIView.animate(withDuration: duration, delay: 0, options: [.curveLinear]) { [weak self] in
+                guard let screenCoordinate = self?.mapView.screenCoordinate(for: location.coordinate) else { return }
+                self?.userCourseView.center = CGPoint(x: screenCoordinate.x, y: screenCoordinate.y)
+            }
+            
+            let cameraOptions = CameraOptions(cameraState: mapView.cameraState)
+            userCourseView.update(location: location,
+                                  pitch: cameraOptions.pitch!,
+                                  direction: cameraOptions.bearing!,
+                                  animated: animated,
+                                  navigationCameraState: navigationCamera.state)
+        case .puck2D(configuration: _):
+            if simulation {
+                if !(mapView.location.locationProvider is PassiveLocationManager) {
+                    mapView.location.locationProvider.stopUpdatingLocation()
+                }
+                if mapView.mapboxMap.style.layerExists(withId: NavigationMapView.LayerIdentifier.puck2DLayer) {
+                    let newLocation: [Double] = [
+                        location.coordinate.latitude,
+                        location.coordinate.longitude,
+                        location.altitude
+                    ]
+                    do {
+                        try mapView.mapboxMap.style.setLayerProperties(for: NavigationMapView.LayerIdentifier.puck2DLayer,
+                                                                       properties: [
+                                                                        "location": newLocation,
+                                                                        "bearing": location.course
+                                                                       ])
+                    } catch {
+                        NSLog("Failed to perform operation while updating 2D puck bearing arrow with error: \(error.localizedDescription).")
```

Comment:
Here’s another use of NSLog.

Reply:
Thanks for the reminder, I'll remove this one.